pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -388,6 +388,8 @@ func (t *ResolvedPipelineTask) skip(facts *PipelineRunFacts) TaskSkipStatus {
 		skippingReason = v1beta1.PipelineTimedOutSkip
 	case t.skipBecausePipelineRunTasksTimeoutReached(facts):
 		skippingReason = v1beta1.TasksTimedOutSkip
+	case t.skipBecauseMatrixParamsContainEmptyArr(facts):
+		skippingReason = v1beta1.TasksTimedOutSkip
```

Comment:
`skippingReason` is set to wrong reason here, set to `TasksTimedOutSkip` instead of `MatrixParamsContainEmptyArrSkip`